### PR TITLE
GHA: use ssh -tt instead of script tool (and fix CI failure reporting)

### PR DESCRIPTION
**Pull request by kolyshkin**
Github Actions execution environments lack tty, which is needed for
integration tests. We have used script utility before to provide a fake
tty, but it appears to be buggy on CentOS 7.
    
Since we use ssh anyway to access Vagant VM, add -tt to forcefully
allocate a tty.
    
Apparently, neither "TERM=xterm" nor "stty size" is needed for
the tests to work, so remove those, too.

The benefits of this are:
 - simpler code;
 - more robust error reporting (I saw failed tests but zero exit code on CentOS 7, presumably an issue with script).

(this also carries some minor cleanups from #2730)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -74,19 +74,19 @@ jobs:
         sudo chown -R rootless.rootless /home/rootless
 
     - name: integration test (fs driver)
-      run: sudo -E PATH="$PATH" script -e -c /bin/bash -c 'stty rows 40 cols 80; TERM=xterm make local${{ matrix.rootless }}integration'
+      run: sudo -E PATH="$PATH" script -e -c 'make local${{ matrix.rootless }}integration'
```
Comment (AkihiroSuda) [inline .github/workflows/test.yml:77]: Do we still need `script`?

Comment (kolyshkin) [reply .github/workflows/test.yml:77]: In this case yes -- there's no terminal otherwise, and there' no `ssh -tt` to make one (although I was thinking about ssh-ing to locahost for the sake of using `-tt`, but `script` looks more elegant).

Also, in this case script is working fine -- I think the only issue is with the one from CentOS 7.

Comment (AkihiroSuda) [reply .github/workflows/test.yml:77]: Please add that to comment lines? Then LGTM

Comment (kolyshkin) [reply .github/workflows/test.yml:83]: Yep, I had this comment and then it was removed.

Added.

**Review by AkihiroSuda — APPROVED** [review]

**Review by mrunalp — APPROVED** [review]
